### PR TITLE
Added BufferSize to Jetsream and NATS outputs

### PR DESCRIPTION
**Pull request by wendall-robinson**
## Primer
This is a follow-up to [THIS](https://github.com/openconfig/gnmic/issues/342) issue.  We would like to set the `buffer-size` field in the JetStream config output when the Go channel is created but this option is not currently available.   We noticed this isn't available in the NATs output either so we added it as well.

## Change
- [x] added BufferSize as a Config field for NATs and JetStream outputs
- [x] added  the BufferSize option to the Go channel when created
- [x] updated the user documents

### CODE REVIEW

File: docs/user_guide/outputs/jetstream_output.md [hunk 1]
```diff
@@ -123,6 +123,10 @@ outputs:
     # boolean, enables extra logging for the nats output
     debug: false
     # boolean, enables the collection and export (via prometheus) of output specific metrics
+    # integer, sets the size of the local buffer where received
+    # NATS messages are stored before being sent to outputs.
+    # This value is set per worker. Defaults to 100 messages
+    buffer-size: 100
```
Comment (karimra) [inline docs/user_guide/outputs/jetstream_output.md:129]: Looks like this was inserted after the comment for `enable-metrics: false`.
Can we also keep the default to 0 to not change the default behavior?
I didn't see the default being set in code anyways.

File: docs/user_guide/outputs/nats_output.md [hunk 2]
```diff
@@ -73,6 +73,10 @@ outputs:
     # boolean, enables extra logging for the nats output
     debug: false
     # boolean, enables the collection and export (via prometheus) of output specific metrics
+    # integer, sets the size of the local buffer where received
+    # NATS messages are stored before being sent to outputs.
+    # This value is set per worker. Defaults to 100 messages
+    buffer-size: 100
```
Comment (karimra) [inline docs/user_guide/outputs/nats_output.md:79]: Same comment as Jetstream

File: pkg/outputs/nats_outputs/jetstream/jetstream_output.go [hunk 3]
```diff
@@ -87,6 +87,7 @@ type config struct {
 	NumWorkers         int                 `mapstructure:"num-workers,omitempty" json:"num-workers,omitempty"`
 	WriteTimeout       time.Duration       `mapstructure:"write-timeout,omitempty" json:"write-timeout,omitempty"`
 	Debug              bool                `mapstructure:"debug,omitempty" json:"debug,omitempty"`
+	BufferSize         int                 `mapstructure:"buffer-size,omitempty"`
```
Comment (karimra) [inline pkg/outputs/nats_outputs/jetstream/jetstream_output.go:90]: Can we change the type to uint? Avoids having to deal with negative numbers and changing types when using it.

File: pkg/outputs/nats_outputs/nats/nats_output.go [hunk 4]
```diff
@@ -87,6 +87,7 @@ type Config struct {
 	NumWorkers         int              `mapstructure:"num-workers,omitempty"`
 	WriteTimeout       time.Duration    `mapstructure:"write-timeout,omitempty"`
 	Debug              bool             `mapstructure:"debug,omitempty"`
+	BufferSize         int              `mapstructure:"buffer-size,omitempty"`
```
Comment (karimra) [inline pkg/outputs/nats_outputs/nats/nats_output.go:90]: Same comment as jetstream.

**Review by karimra — CHANGES_REQUESTED** [review]

File: docs/user_guide/outputs/jetstream_output.md (hunk 1 above)
Comment (wendall-robinson) [reply docs/user_guide/outputs/jetstream_output.md:129]: Certainly, I can make that change

File: pkg/outputs/nats_outputs/jetstream/jetstream_output.go (hunk 3 above)
Comment (wendall-robinson) [reply pkg/outputs/nats_outputs/jetstream/jetstream_output.go]: This has been updated as requested.

Comment (wendall-robinson) [reply pkg/outputs/nats_outputs/jetstream/jetstream_output.go]: This has been updated and fixed as requested.

File: docs/user_guide/outputs/jetstream_output.md (hunk 1 above)
Comment (wendall-robinson) [reply docs/user_guide/outputs/jetstream_output.md]: This has been updated as requested.

File: docs/user_guide/outputs/nats_output.md (hunk 2 above)
Comment (wendall-robinson) [reply docs/user_guide/outputs/nats_output.md]: This has been updated as requested.

File: pkg/outputs/nats_outputs/jetstream/jetstream_output.go [hunk 5]
```diff
@@ -136,7 +137,7 @@ func (n *jetstreamOutput) Init(ctx context.Context, name string, cfg map[string]
 		return err
 	}
 
-	n.msgChan = make(chan *outputs.ProtoMsg)
+	n.msgChan = make(chan *outputs.ProtoMsg, uint(n.Cfg.BufferSize))
```
Comment (karimra) [inline pkg/outputs/nats_outputs/jetstream/jetstream_output.go:140]: this type conversion is not necessary anymore.

File: pkg/outputs/nats_outputs/nats/nats_output.go [hunk 6]
```diff
@@ -145,7 +146,7 @@ func (n *NatsOutput) Init(ctx context.Context, name string, cfg map[string]inter
 		return err
 	}
 
-	n.msgChan = make(chan *outputs.ProtoMsg)
+	n.msgChan = make(chan *outputs.ProtoMsg, uint(n.Cfg.BufferSize))
```
Comment (karimra) [inline pkg/outputs/nats_outputs/nats/nats_output.go:149]: this type conversion is not necessary anymore.

**Review by karimra — CHANGES_REQUESTED** [review]

File: pkg/outputs/nats_outputs/jetstream/jetstream_output.go (hunk 5 above)
Comment (wendall-robinson) [reply pkg/outputs/nats_outputs/jetstream/jetstream_output.go]: Good, catch, this has been updated.

File: pkg/outputs/nats_outputs/nats/nats_output.go (hunk 6 above)
Comment (wendall-robinson) [reply pkg/outputs/nats_outputs/nats/nats_output.go]: Good, catch, this has been updated.

**Review by karimra — APPROVED** [review]

**Comment by karimra** [comment]
Thanks for the contribution!

**Pull request merged**